--- ../node/test/pummel/test-exec.js	2013-08-08 17:10:55.110705338 -0700
+++ test/pummel/test-exec.js	2014-06-27 10:53:52.550470200 -0700
@@ -75,7 +75,7 @@
 exec(SLEEP3_COMMAND, { timeout: 50 }, function(err, stdout, stderr) {
   var diff = (new Date()) - sleeperStart;
   console.log('\'sleep 3\' with timeout 50 took %d ms', diff);
-  assert.ok(diff < 500);
+  assert.ok(diff < 5000);
   assert.ok(err);
   assert.ok(err.killed);
   assert.equal(err.signal, 'SIGTERM');
@@ -106,7 +106,7 @@
 
   // the timeout should still be in effect
   console.log('\'sleep 3\' was already killed. Took %d ms', diff);
-  assert.ok(diff < 1500);
+  assert.ok(diff < 5000);
 }
 
 
