--- ../node/test/pummel/test-net-pause.js	2013-08-08 17:10:55.110705338 -0700
+++ test/pummel/test-net-pause.js	2014-06-27 12:23:52.013231264 -0700
@@ -68,13 +68,13 @@
           assert.equal(chars_recved, recv.length);
           client.resume();
 
-        }, 500);
+        }, 1000);
 
-      }, 500);
+      }, 1000);
 
-    }, 500);
+    }, 1000);
 
-  }, 500);
+  }, 1000);
 
   client.on('end', function() {
     server.close();
