--- ../node/test/simple/test-child-process-fork-dgram.js	2014-09-09 10:48:40.124421023 -0700
+++ test/simple/test-child-process-fork-dgram.js	2014-09-09 11:12:45.423906087 -0700
@@ -41,6 +41,7 @@
 if (process.platform === 'win32') {
   console.error('Sending dgram sockets to child processes not supported');
   process.exit(0);
+  return;
 }
 
 if (process.argv[2] === 'child') {
@@ -103,7 +104,7 @@
         shutdown();
       }
 
-    }, 1);
+    }, 100);
   };
 
   var shutdown = function () {
