--- ../node/test/simple/test-stdout-close-catch.js	2014-06-03 10:30:02.920806600 -0700
+++ test/simple/test-stdout-close-catch.js	2014-06-03 10:28:16.616549766 -0700
@@ -28,9 +28,9 @@
 
 var testScript = path.join(common.fixturesDir, 'catch-stdout-error.js');
 
-var cmd = JSON.stringify(process.execPath) + ' ' +
+var cmd = process.execPath + ' ' +
           JSON.stringify(testScript) + ' | ' +
-          JSON.stringify(process.execPath) + ' ' +
+          process.execPath + ' ' +
           '-pe "process.exit(1);"';
 
 var child = child_process.exec(cmd);
