--- ../node/test/simple/test-signal-handler.js	2013-08-08 18:44:24 -0700
+++ test/simple/test-signal-handler.js	2014-06-06 11:32:36 -0700
@@ -23,10 +23,14 @@
 // SIGUSR1 and SIGHUP are not supported on Windows
 if (process.platform === 'win32') {
   process.exit(0);
+  return;
 }
 
 var common = require('../common');
 var assert = require('assert');
+// Avatar.js specific
+process.signals.start('SIGHUP');
+process.signals.start('SIGUSR1');
 
 console.log('process.pid: ' + process.pid);
 
