--- ../node/test/simple/test-child-process-fork-dgram.js	2013-08-08 18:44:23 -0700
+++ test/simple/test-child-process-fork-dgram.js	2014-06-06 11:33:54 -0700
@@ -27,6 +27,7 @@
 if (process.platform === 'win32') {
   console.error('Sending dgram sockets to child processes not supported');
   process.exit(0);
+  return;
 }
 
 if (process.argv[2] === 'child') {
@@ -86,7 +87,7 @@
           }
         }
       );
-    }, 1);
+    }, 100);
   };
 
   var shutdown = function () {
