--- ../node/test/simple/test-domain.js	2014-06-03 13:36:35.668847852 -0700
+++ test/simple/test-domain.js	2014-06-03 13:49:48.644328757 -0700
@@ -104,7 +104,7 @@
       assert.ok(!er.domainBound);
       break;
 
-    case 'Cannot call method \'isDirectory\' of undefined':
+    case 'Cannot read property "isDirectory" from undefined':
       assert.equal(er.domain, d);
       assert.ok(!er.domainEmitter);
       assert.ok(!er.domainBound);
