--- ../node/test/common.js	2014-09-09 10:48:40.124421023 -0700
+++ test/common.js	2014-09-09 11:07:25.839900413 -0700
@@ -67,7 +67,7 @@
 exports.ddCommand = function(filename, kilobytes) {
   if (process.platform === 'win32') {
     var p = path.resolve(exports.fixturesDir, 'create-file.js');
-    return '"' + process.argv[0] + '" "' + p + '" "' +
+    return process.argv[0] + ' "' + p + '" "' +
            filename + '" ' + (kilobytes * 1024);
   } else {
     return 'dd if=/dev/zero of="' + filename + '" bs=1024 count=' + kilobytes;
@@ -147,7 +147,8 @@
     knownGlobals.push(Uint32Array);
     knownGlobals.push(Float32Array);
     knownGlobals.push(Float64Array);
-    knownGlobals.push(DataView);
+    //nashorn https://jbs.oracle.com/bugs/browse/JDK-8015958
+    //knownGlobals.push(DataView);
   }
 
   for (var x in global) {
@@ -210,3 +211,12 @@
     return fn.apply(this, arguments);
   };
 };
+
+exports.isIPv6Enabled = true;
+var dgram = require('dgram');
+var socket = dgram.createSocket('udp6');
+try {
+    socket.send6('', common.PORT, '::1');
+} catch(error) {
+    exports.isIPv6Enabled = false;
+}
