--- ../node/lib/net.js	2014-06-04 14:14:39.963752412 -0700
+++ src/main/js/lib/net.js	2014-06-04 14:14:54.327740417 -0700
@@ -395,7 +395,7 @@
 Socket.prototype.end = function(data, encoding) {
   stream.Duplex.prototype.end.call(this, data, encoding);
   this.writable = false;
-  DTRACE_NET_STREAM_END(this);
+  //DTRACE_NET_STREAM_END(this);
 
   // just in case we're waiting for an EOF.
   if (this.readable && !this._readableState.endEmitted)
@@ -475,7 +475,7 @@
   fireErrorCallbacks();
 
   if (this.server) {
-    COUNTER_NET_SERVER_CONNECTION_CLOSE(this);
+    //COUNTER_NET_SERVER_CONNECTION_CLOSE(this);
     debug('has server');
     this.server._connections--;
     if (this.server._emitCloseIfDrained) {
@@ -1186,8 +1186,8 @@
   self._connections++;
   socket.server = self;
 
-  DTRACE_NET_SERVER_CONNECTION(socket);
-  COUNTER_NET_SERVER_CONNECTION(socket);
+  //DTRACE_NET_SERVER_CONNECTION(socket);
+  //COUNTER_NET_SERVER_CONNECTION(socket);
   self.emit('connection', socket);
 }
 
