--- ../node/lib/timers.js	2014-06-04 14:14:39.963752412 -0700
+++ src/main/js/lib/timers.js	2014-06-04 14:14:55.587739493 -0700
@@ -433,6 +433,12 @@
   if (!msecs || msecs < 0) return;
   assert(msecs >= 0);
 
+  // nashorn perf optimization
+  // This helps the compiler know that the value is a number
+  // and that type knowledge then propagates to "var when" at line 458
+  // and then also to fast double comparison in "if (when < them)" in the loop
+  msecs = +msecs;
+
   L.remove(item);
 
   if (!unrefList) {
